toolkit/src/main/java/com/onelogin/saml2/Auth.java

```diff
@@ -655,6 +656,13 @@ public void processResponse(String requestId) throws Exception {
 				LOGGER.error("processResponse error. invalid_response");
 				LOGGER.debug(" --> " + samlResponseParameter);
 				errorReason = samlResponse.getError();
+				SamlResponseStatus samlResponseStatus = samlResponse.getResponseStatus();
+				if (samlResponseStatus.getStatusCode() != null) {
+					errors.add(samlResponseStatus.getStatusCode());
```

Comment:
Instead of adding getStatusCode and getSubStatusCode string as errors, why better no modify the checkStatus method to include the getSubStatusCode as part of the statusExceptionMsg?

Similar should be done to the [LogoutResponse](https://github.com/onelogin/java-saml/blob/master/toolkit/src/main/java/com/onelogin/saml2/Auth.java#L701)

Reply:
This has the advantage that: a) it Looks like a stacktrace, and b) you can easily search the list of specific strings (exact match) to know what to show the user.  if it is just part of a bigger string, then you have to search for a string with a string.  This suited our needs better.

I agree about the LogoutResponse.  if you want to accept this way, I can make the same changes there.